### PR TITLE
Carli/2054 support keeping moment image

### LINKED ISSUE #2054: Option to keep or replace generated moment images when a new request is made (frontend)
Follow-up to #1951 for the PV generator dialog, this feature is the option to keep **moment** images.

### LINKED ISSUE #2061: Mobx error in console when opening z profiler without loading images
**Describe the bug**
Console shows an error message when opening z profiler without any images loaded.

**To Reproduce**
1. Launch Carta.
2. Don't load any images and open z profiler.
3. Check console.

**Expected behavior**
No error messages shown.

**Screenshots or videos**
image

**Platform info (please complete the following information):**
 - OS [e.g. macOS Monterey]: macOS
 - Browser [e.g. chrome, safari, electron app]: chrome
 - Browser version [e.g. 22]: 107
 - Backend branch [e.g. dev, v3b2 release]: dev
 - Frontend branch [e.g. dev, v3b2 release]: dev

### PR DESCRIPTION
**Pull request by crocka**
**Description**

Resolves #2054. Added option to keep previously generated moment images.
Resolves #2061.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] ~e2e test passing~ / corresponding fix added
- [x] changelog updated / ~no changelog update needed~
- [x] protobuf updated to the latest dev commit / ~~no protobuf update needed~~
- [x] `BackendService` unchanged / ~~`BackendService` changed and corresponding ICD test fix added~~

### CODE REVIEW

**Comment by kswang1029** [comment]
@pford @crocka The keep function for moment images appears fine. I noticed that the keep function for pv images seems broken. Could you have a look?

**Comment by kswang1029** [comment]
with the new toggle for keep, we need a taller default dialog height so the generate button is visual and there will be no scroll bar present.

**Comment by kswang1029** [comment]
ready for a PR. 

tested manually and automatically with e2e tests. All looking good, except UI changes which break moment generator tests. A branch has been created to address the changes.

File: src/components/SpectralProfiler/MomentGeneratorComponent/MomentGeneratorComponent.tsx [hunk 1]
```diff
@@ -239,6 +239,15 @@ export class MomentGeneratorComponent extends React.Component<{widgetStore: Spec
                         }}
                     />
                 </FormGroup>
+                <Divider />
+                <FormGroup inline={true} label={"Keep previous Moment Image"}>
+                    <Switch
+                        onChange={event => {
+                            const e = event.target as HTMLInputElement;
+                            this.props.widgetStore.setKeep(e.checked);
```
Comment (YuHsuan-Hwang) [inline src/components/SpectralProfiler/MomentGeneratorComponent/MomentGeneratorComponent.tsx]: `this.props.widgetStore` can be simplified to `widgetStore`

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
The changes look good. How about we also fix #2061 in this PR? Shall be a one-line change.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Comment by YuHsuan-Hwang** [comment]
I think we will need to update the change log for issue 2054.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Review by kswang1029 — APPROVED** [review]
all looking good. no regression is seen from e2e tests. A fix to moment generator test is applied due to UI changes.

**Comment by kswang1029** [comment]
also confirmed that #2061 is fixed in this PR. 👍

**Pull request merged**